lua/orgmode/colors/markup_highlighter.lua

```diff
@@ -173,7 +205,16 @@ local function get_matches(bufnr, first_line, last_line)
   for _, match, _ in query:iter_matches(root, bufnr, first_line, last_line) do
     for _, node in pairs(match) do
       local char = node:type()
+      if char == '\\' then
```

Comment:
I think this was a bit of premature optimization. I have reverted most of the additions and simplified the table with matching elements. Rather than '\\(' we can just match on '\\'. 

In the future if we want to add markup for entities like `\cent` or latex fragments like `\newline` we might need to reconsider how to distinguish between those.